platforms/ide/problems-api/src/main/java/org/gradle/api/problems/internal/DefaultProblemBuilder.java

```diff
@@ -267,6 +275,12 @@ public InternalProblemSpec diagnostics(ProblemDiagnostics diagnostics) {
         return this;
     }
 
+    @Override
+    public InternalProblemSpec stackAsOriginLocation() {
```

Comment:
🤔 Why do we need this new method now? Why would we collect a stacktrace that is not the origin location?

Reply:
The only place this is currently true is the deprecations.

How would we even distinguish between origin and contextual?

Reply:
Well, if you call `stackTraceLocation()` then it isn't contextual, since you requested it. That is the current behavior, not sure why you want to change it.

Reply:
Removed `stackTraceLocation`